src/imageio/format/jxl.c

```diff
@@ -0,0 +1,498 @@
+/*
+    This file is part of darktable,
+    Copyright (C) 2013-2021 darktable developers.
+    darktable is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    darktable is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License
+    along with darktable.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "bauhaus/bauhaus.h"
+#include "common/darktable.h"
+#include "common/exif.h"
+#include "common/imageio.h"
+#include "control/conf.h"
+#include "imageio/format/imageio_format_api.h"
+
+#include "jxl/encode.h"
+#include "jxl/resizable_parallel_runner.h"
+#include "jxl/thread_parallel_runner.h"
+
+DT_MODULE(1)
+
+
+
+#define CONF_PREFIX "plugins/imageio/format/jxl/"
+
+// Should the decoding effort option be exposed to user?
+//#define SHOW_DECODING_EFFORT
```

Comment:
I was unsure whether to remove this or not. 
Do you think it would be a useful feature to be able to change how quickly exported jxl images can be decoded?
I do not want to overcomplicate the settings for the export, but perhaps this is valuable enough...

Reply:
We should probably decide now to either have the option, or not have it at all (and fix the behaviour for this version), rather than have it at compile time and risk having databases w/ and w/o the field between versions and builds. It could also be introduced as well later and module/database version bumped then...

What does this impact, encoding time/effort as well? If this is significant, I could see why someone would care to have it if they're doing a large batch export for example... If, not, why wouldn't you want to have faster decoding?

Reply:
> What does this impact, encoding time/effort as well
I think the side-effects is the outputted image size (trade-off between compression and decoding speed) but this is not documented.
I will add decoding speed option in and remove the pre-compile flag as long as you are not particularly concerned with an increased complexity.